lib/src/iterable_extensions.dart

```diff
@@ -589,6 +589,19 @@ extension IterableExtension<T> on Iterable<T> {
     }
     return true;
   }
+
+  /// The elements with [seperator] between each.
+  Iterable<T> seperatedBy(T seperator) sync* {
```

Comment:
This one feels pretty specialized. I'm not sure if there will be wide enough usage to have this in a core package.

Reply:
It worries me a little too. I have seen other versions of it in wild, one called `interspersed` and used to insert padding between Flutter widgets in a list.
It's basically the `join` function separator, without the actual joining.

I'm guessing one use-case could be something where you do a `for`-`in` on the result, and treat the separators specially, so you can "functionally" handle the fence-post problem.

So yes, I do want to hear use-cases for `separatedBy` before agreeing to it.